### PR TITLE
Allow PCA to run with given reference mean values

### LINKED ISSUE #2728: PCA test sometimes fails...
I get [this test](https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/analysis/test_pca.py#L166) failing locally when I run in parallel:

```
=============================================================== FAILURES ===============================================================
___________________________________________________________ test_given_mean ____________________________________________________________
[gw0] linux -- Python 3.6.10 /home/richard/miniconda3/envs/mda/bin/python

pca_aligned = <MDAnalysis.analysis.pca.PCA object at 0x7f7cda848588>, u = <Universe with 3341 atoms>

    def test_given_mean(pca_aligned, u):
        pca = PCA(u, select=SELECTION, align=False,
                  mean=pca_aligned._mean).run()
>       assert_almost_equal(pca_aligned.cov, pca.cov, decimal=5)
E       AssertionError: 
E       Arrays are not almost equal to 5 decimals
E       
E       Mismatched elements: 900 / 900 (100%)
E       Max absolute difference: 1.0380358
E       Max relative difference: 106.43334522
E        x: array([[ 6.85943e-02,  6.14962e-03, -2.85235e-02, -2.46964e-03,
E                4.93089e-03, -1.45766e-02,  6.77579e-03,  6.73560e-03,
E               -6.87083e-03, -1.39547e-02,  4.34639e-03, -1.46411e-02,...
E        y: array([[ 1.10663e+00, -4.76013e-01,  2.59657e-01,  8.86885e-01,
E               -4.62178e-01,  3.13851e-01,  7.73986e-01, -4.43201e-01,
E                2.00352e-01,  4.87414e-01, -3.32454e-01,  1.01439e-01,...

analysis/test_pca.py:169: AssertionError
```

This is when I run `pytest -n 12` from the MDAnalysisTests/ directory.  If I run `pytest -n 12 analysis/` it doesn't fail any more.  Varying core count seems to make it fail/not fail, so it feels like something unthreadsafe is happening?

I've tried reducing the scope of the fixture and this still fails.

### PR DESCRIPTION
**Pull request by fiona-naughton**
Fixes #2728

Changes made in this Pull Request:
 - Fix shape of positions array from provided reference atomgroup so the PCA will actually run
 - Update the corresponding test to actually run with given mean values

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_pca.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_pca.py:170]: according to the docs we should just be able to use `mean_atoms` back as the atomgroup with the mean positions right?

Comment (fiona-naughton) [reply testsuite/MDAnalysisTests/analysis/test_pca.py:170]: From the current docs, yes - but, per discussion in #3285, it seems that `mean_atoms` does not actually contain the mean positions. If we fix that, then we could just use `mean_atoms` here;  I decided to do it roundabout like this for now with a view of getting the more 'urgent' changes merged rather than debating exactly what `mean_atoms` should be doing (imo the doc changes in #3285 reflect what `mean_atoms` seem to actually be storing atm so it's not a big after that goes through). 

All that being said, there is still the option to fix `mean_atoms` here (or in 3285), if that seems better? (Personally, it feels like both `means` and `mean_atoms` would be better as position arrays rather than atomgroups, but I'm not super familiar with pca and how people use this part of the input/output so others should weigh in first)

Comment (IAlibay) [reply testsuite/MDAnalysisTests/analysis/test_pca.py:170]: I'd vote for just fixing it properly now (as a coordinate array) and not going through a deprecation cycle.

Comment (orbeckst) [reply testsuite/MDAnalysisTests/analysis/test_pca.py:170]: Works for me.

**Comment by fiona-naughton** [comment]
`mean_atoms` has been removed and I switched `mean` to an array. I decided against allowing atomgroups to be passed in for now and depreciating, since part of the original error means that setting `mean` didn't work anyway, so that implies people weren't using it anyway (without patching the code themselves) - but I'm open to if others think it's better we do.

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/pca.py:225]: pep 8: space after comma
```suggestion
            self.mean = np.zeros((self._n_atoms, 3))
```

File: testsuite/MDAnalysisTests/analysis/test_pca.py
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_pca.py:174]: pep 8
```suggestion
    wrong_mean = [[0,0,0], [1,1,1]]
```

**Review by orbeckst — APPROVED** [review]
Excellent!

(I'm just fixing 2 pep8 things)

**Comment by orbeckst** [comment]
@IAlibay I'll merge this unless you object or CI is unhappy.

**Comment by orbeckst** [comment]
Sigh, not sure what sphinx is unhappy about:
```
/home/runner/work/mdanalysis/mdanalysis/package/MDAnalysis/analysis/pca.py:docstring of MDAnalysis.analysis.pca.PCA.transform:5:Inline literal start-string without end-string.
```

File: package/MDAnalysis/analysis/pca.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/pca.py:321]:
```suggestion
            Include every `step` frames in the PCA transform. If set to
            ``None`` (the default) then every frame is analyzed (i.e., same as
            ``step=1``).
```

I'm running out weird movie-based PEP8 messages to throw.

File: testsuite/MDAnalysisTests/analysis/test_pca.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_pca.py:174]:
```suggestion
    wrong_mean = [[0, 0, 0], [1, 1, 1]]
```
One does not simply PEP8 into M(ordor)DA?

File: testsuite/MDAnalysisTests/analysis/test_pca.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_pca.py:177]:
```suggestion
    with pytest.raises(ValueError, match='Number of atoms in'):
        pca = PCA(u, select=SELECTION, mean=wrong_mean).run()
```
> Hello there
General Ken(pep8)-Obi!

(re-structuring because the indentations are don't match up and this is kinda neater)

**Review by IAlibay — APPROVED** [review]
Overall lgtm thanks @fiona-naughton, please do merge in the following PEP8 fixes

(I lack sleep, so now all PEP8 messages are movie meme-based).

**Comment by orbeckst** [comment]
Thank you @fiona-naughton for fixing this one!

**Pull request merged**
